docs/csharp/tutorials/upgrade-to-nullable-references.md

```diff
@@ -0,0 +1,153 @@
+
+There's quite a bit going on in the preceding block of code. This application uses the [AutoMapper](http://automapper.org/) NuGet package to construct a news item from an `ISyndicationItem`. You've discovered that the news story items are constructed and the properties are set in that one statement. That means the design for the `NewsStoryViewModel` indicates that these properties should never have the `null` value. These properties should be **nonnullable reference types**. That best expresses the original design intent. In fact, any `NewsStoryViewModel` *is* correctly instantiated with non-null values. That makes the following initialization code a valid fix:
+
+```csharp
+public class NewsStoryViewModel
+{
+    public DateTimeOffset Published { get; set; }
+    public string Title { get; set; } = default!;
+    public string Uri { get; set; } = default!;
+}
+```
+
+The assignment of `Title` and `Uri` to `default` which is `null` for the `string` type doesn't change the runtime behavior of the program. The `NewsStoryViewModel` is still constructed with null values, but now the compiler reports no warnings. The **null forgiveness operator**, the `!` character following the `default` expression tells the compiler that the preceding expression is not null. This technique may be expedient when other changes force much larger changes to a code base, but in this application there is a relatively quick and better solution: Make the `NewsStoryViewModel` an immutable type where all the properties are set in the constructor. Make the following changes to the `NewsStoryViewModel`:
+
+[!code-csharp[FinishedViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/ViewModels/NewsStoryViewModel.cs#FinishedViewModel)]
+
+Once that's done, you need to update the code that configures the automapper so that it uses the constructor rather than setting properties. Open `NewsService.cs` and look for the following code at the bottom of the file:
+
+[!code-csharp[StarterAutoMapper](../../../samples/csharp/tutorials/nullable-reference-migration/start/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+That code maps properties of the `ISyndicationItem` object to the `NewsStoryViewModel` properties. You want the AutoMapper to provide the mapping using a constructor instead. Replace the above code with the following automapper configuration:
+
+[!code-csharp[FinishedViewModel](../../../samples/csharp/tutorials/nullable-reference-migration/finished/SimpleFeedReader/Services/NewsService.cs#ConfigureAutoMapper)]
+
+Notice that because this class is small, and you've examined carefully, you should turn on the `#nullable enable` directive above this class declaration. This change could have broken something, so it's worthwhile to run all the tests and test the application before moving on.
```

Comment:
Possibly clarify that "this change" refers to the constructor change, not using `#nullable enable`? It would be good to hammer home the message of "no code generation is changed by nullable contexts".